### PR TITLE
test(sign-up): testing component

### PR DESCRIPTION
**Pull request by YgRamos9**
## Summary
- adding unit tests

## Trello Card
https://trello.com/c/zVbA2x2X/17-tests-de-sign-up

### CODE REVIEW

File: angular/src/app/screens/sign-up/sign-up.component.spec.ts [hunk 1]
```diff
@@ -1,24 +1,87 @@
...
 import { SignUpComponent } from './sign-up.component';
+import { APP_BASE_HREF } from '@angular/common';
 
-describe('SignUpComponent', () => {
-  let component: SignUpComponent;
-  let fixture: ComponentFixture<SignUpComponent>;
+import { fireEvent, render, waitFor } from '@testing-library/angular';
+import { screen } from '@testing-library/dom';
+import '@testing-library/jest-dom';
+import { SIGN_UP_FIELDS } from '../../constants/formAccount';
```
Comment (miguelhenaoa) [inline angular/src/app/screens/sign-up/sign-up.component.spec.ts]: Sort imports

File: angular/src/app/screens/sign-up/sign-up.component.spec.ts [hunk 2]
```diff
@@ -1,24 +1,87 @@
...
+      target: {
+        value: 'ramos111com12',
+      },
+    });
+    await waitFor(() => fireEvent.click(screen.getByText('Sign Up')));
+
+    expect(
+      screen.getByText('El email es requerido y debe ser válido'),
+    ).toBeInTheDocument();
+
```
Comment (miguelhenaoa) [inline angular/src/app/screens/sign-up/sign-up.component.spec.ts]: Remove blank line

File: angular/src/app/screens/sign-up/sign-up.component.spec.ts [hunk 3]
```diff
@@ -1,24 +1,87 @@
...
 
-  it('should create', () => {
-    expect(component).toBeTruthy();
+  it('some wrong field shows an error and does not send the form', async () => {
+    const email = screen.getByLabelText(/Email/i);
+    fireEvent.input(email, {
+      target: {
+        value: 'ramos111com12',
+      },
+    });
```
Comment (miguelhenaoa) [inline angular/src/app/screens/sign-up/sign-up.component.spec.ts]: Use addValueEvent

File: angular/src/app/screens/sign-up/sign-up.module.ts [hunk 4]
```diff
@@ -8,11 +8,6 @@ import { FormContainerComponent } from 'src/app/components/form-container/form-c
 @NgModule({
   declarations: [SignUpComponent, FormContainerComponent],
   imports: [CommonModule, SignUpRoutingModule, ReactiveFormsModule],
-  exports: [
-    SignUpComponent,
-    FormContainerComponent,
-    CommonModule,
-    ReactiveFormsModule,
-  ],
+  exports: [SignUpComponent, FormContainerComponent, CommonModule],
```
Comment (miguelhenaoa) [inline angular/src/app/screens/sign-up/sign-up.module.ts]: It's not necessary to export components if you not using them in an external module

File: angular/src/app/screens/sign-up/sign-up.component.spec.ts [hunk 5]
```diff
@@ -1,24 +1,87 @@
...
+      'Confirmación de Password',
+    );
+    addValueEvent(userName, 'Yonathan Ramos');
+    addValueEvent(lastName, 'Ramos');
+    addValueEvent(email, 'ramos@angular.com');
+    addValueEvent(password, 'Admin1234');
+    addValueEvent(passwordConfirmation, 'Admin1234');
+    await waitFor(() => fireEvent.click(screen.getByText('Sign Up')));
+
+    expect(createUserMock).toHaveBeenCalled();
```
Comment (miguelhenaoa) [inline angular/src/app/screens/sign-up/sign-up.component.spec.ts:72]: Validate if the same data you enter are being sent

**Review by miguelhenaoa — CHANGES_REQUESTED** [review]
Great job!!

File: angular/src/app/screens/sign-up/sign-up.component.spec.ts [hunk 6]
```diff
@@ -1,24 +1,87 @@
...
-  beforeEach(() => {
-    fixture = TestBed.createComponent(SignUpComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
+  it('check all empty fields', async () => {
+    const form = screen.getByRole('form');
+    fireEvent.submit(form);
+    const errorSpans = screen.getAllByTestId('error-span');
+
+    expect(errorSpans).toHaveLength(5);
```
Comment (joacomiguens) [inline angular/src/app/screens/sign-up/sign-up.component.spec.ts]: I prefer to map SIGN_UP_FIELDS errorSpans and check errorSpans variable includes them 😄

File: angular/src/app/services/user.service.spec.ts [hunk 7]
```diff
@@ -1,12 +1,14 @@
...
 import { UserService } from './user.service';
 
 describe('UserService', () => {
   let service: UserService;
 
   beforeEach(() => {
-    TestBed.configureTestingModule({});
+    TestBed.configureTestingModule({
+      imports: [HttpClientModule],
+    });
```
Comment (joacomiguens) [inline angular/src/app/services/user.service.spec.ts]: use Jest instead of Karma.

File: angular/src/app/app.component.spec.ts [hunk 8]
```diff
@@ -15,19 +15,4 @@ describe('AppComponent', () => {
     const app = fixture.componentInstance;
     expect(app).toBeTruthy();
   });
```
Comment (joacomiguens) [inline angular/src/app/app.component.spec.ts]: Use jest in the file instead of Karma

File: angular/src/app/interfaces/global.interface.ts [hunk 9]
```diff
@@ -4,3 +4,12 @@ export interface IFields {
   type: string;
   errorMessage: string;
 }
+export interface IUser {
+  id: number;
+  firstName: string;
+  lastName: string;
+  email: string;
+  password: string;
+  password_confirmation: string;
```
Comment (joacomiguens) [inline angular/src/app/interfaces/global.interface.ts]: should be passwordConfirmation

**Review by miguelhenaoa — APPROVED** [review]

**Review by joacomiguens — APPROVED** [review]
Great Job!! Just a reminder, please upload code-coverage screenshot on PR description 🎸

**Pull request merged**